lib/index.ts

```diff
@@ -1507,6 +1606,26 @@ export class Worker {
 			return null;
 		}
 
+		// Avoid infinite loop
+		if (
+			insertedContract.type.split('@')[0] === 'action-request' &&
+			insertedContract.data.action === 'action-evaluate-triggers@1.0.0'
+		) {
+			logger.debug(
+				logContext,
+				'Enqueuing but not running commit logic because contract is an action-request for action-evaluate-triggers@1.0.0',
+			);
+			await enqueue(
```

Comment:
It's a bit weird that this is basically replicating the `enqueue` call at the bottom of the `commit` function.
I think it might be neater to put the logic together, and move the trigger evaluation action logic to the bottom as well.
Then you can have an if statement, where you check for the infinite loop, and then enqueue afterwards.

Reply:
Refactored it